packages/react/src/search-box/SearchBoxInput.tsx

```diff
@@ -23,24 +30,61 @@ export type SearchBoxInputProps = BaseSearchBoxInputProps & {
 
 export const SearchBoxInput = forwardRef<HTMLInputElement, SearchBoxInputProps>(
 	function SearchBoxInput(
-		{ label = 'Search', labelVisible = false, id, ...props },
+		{
+			id,
+			label = 'Search',
+			labelVisible = false,
+			onChange: onChangeProps,
+			value: valueProp,
+			...props
+		},
 		ref
 	) {
+		const internalRef = useRef<HTMLInputElement>(null);
+		const [internalValue, setInternalValue] = useState(valueProp || '');
+
+		const value = typeof valueProp === 'string' ? valueProp : internalValue;
+		const showClearButton = Boolean(value);
+
+		const onChange = (event: ChangeEvent<HTMLInputElement>) => {
+			onChangeProps?.(event);
+			setInternalValue(event.target.value);
+		};
+
+		// Clears the input while also triggering the `onChange` event to consumers
+		const clearInput = () => {
+			if (!internalRef.current?.value) return;
+			internalRef.current.value = '';
+			// A synthetic copy of the ChangeEvent<HTMLInputElement>, can't be manually trigger
+			onChange({
+				bubbles: true,
+				cancelable: false,
+				defaultPrevented: false,
+				isTrusted: true,
+				target: internalRef.current,
+				type: 'change',
+			} as ChangeEvent<HTMLInputElement>);
+			internalRef.current?.focus();
+		};
```

Comment:
I think this works and is the correct way of doing things:


```suggestion
		const clearInput = () => {
			if (!internalRef.current?.value) return;
			internalRef.current.value = '';

			const event = new Event('input', { bubbles: true });
			internalRef.current.dispatchEvent(event);

			internalRef.current?.focus();
		};
```

Reply:
Interesting magic and I think it's the way to go. I don't understand how it works, but it seems to!

Reply:
[Heres the source with the link to the proper credits](https://stackoverflow.com/questions/23892547/what-is-the-best-way-to-trigger-change-or-input-event-in-react-js/46012210#46012210)

Basically React overwrites the setter, and we have to apply and use the native setter before sending an event with an updated value. A really smart workaround! (whether or not we should be doing it with React is another thing)

Reply:
It's cooooool 🚢